packages/komencikit/src/client.spec.ts

```diff
@@ -0,0 +1,158 @@
+import * as t from 'io-ts'
+import 'jest-fetch-mock'
+import { action } from './actions'
+import { KomenciClient } from './client'
+import { FetchErrorTypes } from './errors'
+
+const URL = 'http://komenci.celo.com/'
+
+// Quite test output
+jest.spyOn(console, 'debug').mockImplementation(() => null)
+
+describe('KomenciClient', () => {
+  describe('#exec', () => {
+    const TestResp = t.type({
+      field: t.string,
+    })
+    type TestResp = t.TypeOf<typeof TestResp>
+    const testAction = action<'testAction', {}, TestResp>(
+      'testAction',
+      'POST',
+      'testAction',
+      TestResp
+    )
+
+    let client: KomenciClient
+
+    beforeEach(() => {
+      client = new KomenciClient(URL)
+      fetchMock.resetMocks()
+    })
+
+    describe('when the request encounters network errors', () => {
+      it('returns a wrapped error', async () => {
+        const act = testAction({})
+        fetchMock.mockAbort()
+
+        const res = await client.exec(act)
+        expect(res.ok).toBe(false)
+        if (res.ok === false) {
```

Comment:
line 38 will throw if this is not false, right?

Reply:
Yeah exactly, but the reason for the `if` is that the typescript compiler doesn't yet have jest expect comprehension sadly (it's a work in progress thing) so even if I add the expectation for false, the compiler doesn't yet discriminate between the `res` union types unless I do the explicit `if` then I can actually access `res.error`.

I was working on some custom jest matchers to help with our `Result` type but I found it a bit weird to type those so I gave up temporarily.

Reply:
Ah gotcha, makes sense.